src/class/video/video_device.c

```diff
@@ -759,6 +817,7 @@ static uint_fast16_t _prepare_in_payload(videod_streaming_interface_t *stm)
   if (hdr_len + remaining < pkt_len) {
     pkt_len = hdr_len + remaining;
   }
+  TU_ASSERT(pkt_len >= hdr_len);
```

Comment:
while streaming and got probe/commit request: ep_buf is wiped out causing
- hdr_len is incorrect
- max_payload_transfer_size is temporarily cleared as well by _init_vs_configuration() (see below) therefore cause incorrect data_len e,g data_len = 0 - 1 = MAX_UINT16 and overflow the memcpy which cause hardfault.